### ISSUE TITLE
Plot resolution problem when using fig.asp and rendering R Notebook (Retina display)

### ISSUE THREAD

**Issue by gregrs-uk** [rstudio/rstudio#2098]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.419
    OS Version      : OS X 10.11.6 (on a MacBook Pro Retina 13-inch, Early 2015)
    R Version       : 3.4.3 (2017-11-30)

### Steps to reproduce the problem

1. File > New File > R Notebook.

2. Replace the contents of the file with the following, removing the backslashes. (Sorry, I had to add these to prevent GitHub from messing up the formatting.)

```
---
output: html_notebook
---

Here is a plot.

\```{r}
plot(cars)
\```

Here is the same plot with `fig.asp` set.

\```{r fig.asp = 1}
plot(cars)
\```
```

3. Save the file with an `.Rmd` extension.

4. Code > Run Region > Run All

### Describe the problem in detail

Here's the first plot.

![plot1](https://user-images.githubusercontent.com/7353938/35647869-d9540250-06cb-11e8-955d-81ad66cb6c3a.png)

Here's the second plot.

![plot2](https://user-images.githubusercontent.com/7353938/35647878-e57634d6-06cb-11e8-92ee-0d2bd1da6f67.png)

The text and points in the second plot are very small compared with the first plot.

### Describe the behavior you expected

Here's what the second plot looks like when the page is rendered using `rmarkdown::render` in command-line R.

![plot2-correct](https://user-images.githubusercontent.com/7353938/35647939-121a617e-06cc-11e8-898c-7217eca24270.png)

I've also noticed problems when using `fig.width` and `fig.height`. Behaviour has been different when I drag RStudio on to a second monitor so perhaps the issue is caused by my MacBook's retina display.

There's still a problem when viewing using the 'Preview' button, but not when knitting to HTML.

**Comment by ronblum** [comment]
@gregrs-uk Thank you for raising the issue, and I will mark it as a bug for review. In the meantime, one workaround is to adjust the size with cex—e.g. `par(cex = 2.5)`—which will increase the size within the preview. However, you'll need to remove it when knitting the document.

**Comment by gregrs-uk** [comment]
Thanks for the workaround, Ron.

**Comment by gregrs-uk** [comment]
Looks like this is possibly related to #4521

**Comment by eleozzr** [comment]
But what if ggplot? how to fix it?

**Comment by mikebessuille** [comment]
We see quite a few issues related to plot resolution, so I don't want this issue to go stale.  We'll try to look at it in 1.6

**Comment by megaraptor1** [comment]
This issue is still going on, at least for me. Plots in the Notebook are very small, but knitting the whole document displays them at their proper resolution. It's to the point where if I want to see the plot I have to knit the whole document.

**Comment by ronblum** [comment]
Moving to Cherry Blossom, in order to verify that it's fixed by PR #12324. If it's not, we'll reassign it to the current milestone.

**Comment by jonvanausdeln** [comment]
Verified with 2023.03.0-daily+280.  The text and points of the plots are the same in each plot as expected.
